src/server/Monte.AuthServer/Services/UserService.cs

```diff
@@ -0,0 +1,188 @@
+        };
+
+        var result = await _userManager.CreateAsync(user);
+        if (!result.Succeeded)
+        {
+            return new BadRequestObjectResult(result.Errors.ToErrorDict());
+        }
+
+        try
+        {
+            ThrowIfError(await _userManager.AddPasswordAsync(user, request.Password));
+            ThrowIfError(await _userManager.AddToRoleAsync(user, role));
+        }
+        catch (IdentityErrorException e)
+        {
+            await _userManager.DeleteAsync(user);
+            return new BadRequestObjectResult(e.Errors);
+        }
+        catch
+        {
+            await _userManager.DeleteAsync(user);
+            throw;
+        }
+        return new CreatedResult(string.Empty, new UserDetails() { Id = user.Id, Name = user.UserName });
+    }
+    
+    public async Task<IActionResult> ChangeUsername(string userId, string newUsername)
+    {
+        if (string.IsNullOrEmpty(newUsername) || string.IsNullOrEmpty(userId))
+        {
+            return new BadRequestObjectResult("Invalid credentials.");
+        }
+
+        var existingUser = await _userManager.FindByNameAsync(newUsername);
+        if (existingUser is not null)
+        {
+            return new BadRequestObjectResult("This username is already taken.");
+        }
+
+        var usr = await _userManager.FindByIdAsync(userId);
+        if (usr == null)
+        {
+            return new NotFoundObjectResult($"User with the id '{userId}' was not found.");
+        }
+
+
+        try
+        {
+            ThrowIfError(await _userManager.SetUserNameAsync(usr, newUsername));
+        }
+        catch (IdentityErrorException e)
+        {
+            return new BadRequestObjectResult(e.Errors);
+        }
+        catch
+        {
+            throw;
+        }
+
+        return new OkObjectResult(new UserDetails() { Id = usr.Id, Name = usr.UserName ?? "-" });
+    }
+    
+    public async Task<IActionResult> ChangePassword(string userId, string oldPassword, string newPassword)
+    {
+        if (string.IsNullOrEmpty(userId) ||
+            string.IsNullOrEmpty(oldPassword) ||
+            string.IsNullOrEmpty(newPassword))
+        {
+            return new BadRequestObjectResult("Invalid credentials.");
+        }
+
+        var usr = await _userManager.FindByIdAsync(userId);
+        if (usr == null)
+        {
+            return new NotFoundObjectResult($"User with the id '{userId}' was not found.");
+        }
+
+        try
+        {
+            ThrowIfError(await _userManager.ChangePasswordAsync(usr, oldPassword, newPassword));
+        }
+        catch (IdentityErrorException e)
+        {
+            return new BadRequestObjectResult(e.Errors);
+        }
+        catch
+        {
+            throw;
+        }
+
+        return new NoContentResult();
+    }
+
+    public async Task<IActionResult> GetUsers()
+    {
+        var users = await _userManager.GetUsersInRoleAsync(AuthConsts.Roles.MonteUser);
```

Comment:
this will not return the admin, but it should, right?

Reply:
That's actually up to debate. I thought that only the admin will be able to see user list and since there is only one admin he doesn't need to see himself. But if he should i will him to the list,